### PR TITLE
openshift-ci: install with kata-deploy

### LINKED ISSUE #5547: openshift-ci: install with kata-deploy
**Which feature do you think can be improved?**

The daily kata containers CI job on openshift-ci is broken because it cannot build some pieces of the stack anymore. The recent changes on CI scripts that leverage the kata-deploy build scripts is incompatible with the openshift-ci builds environment.

**How can it be improved?**

We have built a payload for kata-deploy for every pull request merged. So the idea is to leverage that payload along side kata-deploy to install kata containers in the test cluster. This way it won't need to build the stack on openshift-ci.

### PR DESCRIPTION
**Pull request by wainersm**
This replaces the legacy scripts to build and install Kata Containers on the test cluster of openshift-ci. Instead it is used kata-deploy and its kata-deploy-ci image payload that is rebuilt for every pull request merged.

Fixes #5547
Signed-off-by: Wainer dos Santos Moschetta <wainersm@redhat.com>

----

I tested it locally with OCP 4.11 and it worked out:
```
$ ./test.sh 
Client Version: 4.12.0-202303081116.p0.g846602e.assembly.stream-846602e
Kustomize Version: v4.5.7
Server Version: 4.11.0-0.nightly-2023-03-09-142909
Kubernetes Version: v1.24.6+deccab3
INFO: Install and configure kata into the test cluster
Now using project "default" on server "https://api.ci-ln-kbg83dt-1d09d.ci.azure.devcluster.openshift.com:6443".
~/go/src/github.com/kata-containers/kata-containers ~/go/src/github.com/kata-containers/tests/.ci/openshift-ci
INFO: Applying kata-deploy
serviceaccount/kata-label-node created
clusterrole.rbac.authorization.k8s.io/node-labeler created
clusterrolebinding.rbac.authorization.k8s.io/kata-label-node-rb created
Warning: would violate PodSecurity "restricted:latest": privileged (container "kube-kata" must not set securityContext.privileged=true), allowPrivilegeEscalation != false (container "kube-kata" must set securityContext.allowPrivilegeEscalation=false), unrestricted capabilities (container "kube-kata" must set securityContext.capabilities.drop=["ALL"]), restricted volume types (volumes "crio-conf", "containerd-conf", "kata-artifacts", "dbus", "systemd", "local-bin" use restricted volume type "hostPath"), runAsNonRoot != true (pod or container "kube-kata" must set securityContext.runAsNonRoot=true), seccompProfile (pod or container "kube-kata" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
daemonset.apps/kata-deploy created
pod/kata-deploy-882q4 condition met
pod/kata-deploy-kpvrx condition met
pod/kata-deploy-vv4dk condition met
INFO: Adding the kata runtime classes
runtimeclass.node.k8s.io/kata-qemu created
runtimeclass.node.k8s.io/kata-clh created
runtimeclass.node.k8s.io/kata-fc created
runtimeclass.node.k8s.io/kata-dragonball created
~/go/src/github.com/kata-containers/tests/.ci/openshift-ci
INFO: Configuring SELinux
machineconfig.machineconfiguration.openshift.io/51-kata-selinux created
NAME              GENERATEDBYCONTROLLER   IGNITIONVERSION   AGE
51-kata-selinux                           2.2.0             0s
Wait ci-ln-kbg83dt-1d09d-jcpgt-worker-centralus1-sqdtm reboot
Wait ci-ln-kbg83dt-1d09d-jcpgt-worker-centralus2-mf5mk reboot
Wait ci-ln-kbg83dt-1d09d-jcpgt-worker-centralus3-dp9vt reboot
Set timeout to 900 seconds
Checking after 60 seconds
Checking after 123 seconds
Checking after 186 seconds
ci-ln-kbg83dt-1d09d-jcpgt-worker-centralus1-sqdtm rebooted
Checking after 254 seconds
Checking after 316 seconds
Checking after 378 seconds
Checking after 441 seconds
ci-ln-kbg83dt-1d09d-jcpgt-worker-centralus2-mf5mk rebooted
Checking after 503 seconds
Checking after 564 seconds
Checking after 625 seconds
Checking after 687 seconds
ci-ln-kbg83dt-1d09d-jcpgt-worker-centralus3-dp9vt rebooted
INFO: Run test suite: smoke
INFO: Creating the http-server pod
Warning: would violate PodSecurity "restricted:latest": allowPrivilegeEscalation != false (container "http-server" must set securityContext.allowPrivilegeEscalation=false), unrestricted capabilities (container "http-server" must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (pod or container "http-server" must set securityContext.runAsNonRoot=true), seccompProfile (pod or container "http-server" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
pod/http-server created
INFO: Wait until the pod gets running
INFO: http-server is running
INFO: Creating the service and route
service/http-server-service created
route.route.openshift.io/http-server-route created
INFO: HTTP server is working
INFO: Deleting resources created
service "http-server-service" deleted
route.route.openshift.io "http-server-route" deleted
INFO: Deleting the http-server pod
pod "http-server" deleted
INFO: Test suite: smoke: PASS
```

### CODE REVIEW

**Comment by wainersm** [comment]
I still want to test it with some other changes on openshift-ci side, so leaving the do-not-merge label.

**Comment by wainersm** [comment]
@fidencio I was thinking in create a `tools/packaging/kata-deploy/kata-deploy/base/kata-deploy-ci.yaml` that would contain the url to the kata-deploy-ci image. Not just openshift-ci but developers would benefit of it, in case they want to install the latest code on their clusters. What do you think?

**Comment by wainersm** [comment]
The pp64le failure is unrelated with this PR:
```
Build and install gperf version 3.1
Failed at 596: ${script_dir}/../../../ci/install_libseccomp.sh "${libseccomp_install_dir}" "${gperf_install_dir}"
make: *** [Makefile:96: /home/jenkins/workspace/kata-containers-2.0-tests-ppc64le-containerd-k8s-ubuntu-20-04-PR/go/src/github.com/kata-containers/kata-containers/tools/packaging/kata-deploy/local-build/build/rootfs-initrd/builddir/initrd-image/.ubuntu_rootfs.done] Error 18
make[1]: *** [tools/packaging/kata-deploy/local-build/Makefile:40: rootfs-initrd-tarball-build] Error 2
make[1]: Leaving directory '/home/jenkins/workspace/kata-containers-2.0-tests-ppc64le-containerd-k8s-ubuntu-20-04-PR/go/src/github.com/kata-containers/kata-containers'
make: *** [tools/packaging/kata-deploy/local-build/Makefile:67: rootfs-initrd-tarball] Error 2
```
The error log is also not clear about the problem...

File: .ci/openshift-ci/cluster/install_kata.sh [hunk 1]
```diff
@@ -27,17 +27,31 @@ KATA_WITH_SYSTEM_QEMU=${KATA_WITH_SYSTEM_QEMU:-no}
...
 #
-export DAEMONSET_LABEL="kata-deploy"
+apply_kata_deploy() {
+	local deploy_file="tools/packaging/kata-deploy/kata-deploy/base/kata-deploy.yaml"
+	local old_img="quay.io/kata-containers/kata-deploy:latest"
+	# Use the kata-deploy CI image which is built for each pull request merged
+	local new_img="quay.io/kata-containers/kata-deploy-ci:kata-containers-latest"
+
+	pushd "$katacontainers_repo_dir"
+	sed -i "s#${old_img}#${new_img}#" "$deploy_file"
```
Comment (snir911) [inline .ci/openshift-ci/cluster/install_kata.sh:43]: would be possible using kustomize for this?

**Review by snir911 — APPROVED** [review]
LGTM thanks! (add one minor comment)

**Review by fidencio — APPROVED** [review]
lgtm, thanks @wainersm!

I'm with @snir911 and I think the more kustomizations we add, the better, so we don't actually have to create new files.

**Pull request merged**